src/core/shards/wires.cpp

```diff
@@ -1761,6 +1761,57 @@ struct StepMany : public TryMany {
   }
 };
 
+struct DoMany : public TryMany {
+  static inline Parameters _params{
+      {"Wire", SHCCSTR("The wire to run many times sequentially."), WireBase::WireVarTypes},
+  };
+
+  static SHParametersInfo parameters() { return _params; }
+
+  SHVar activate(SHContext *context, const SHVar &input) {
+    auto len = getLength(input);
+
+    // Compute required size
+    size_t capacity = std::max(MinWiresAdded, len);
+
+    // Only ever grow
+    capacity = std::max(_wires.size(), capacity);
+
+    _outputs.resize(capacity);
+    _wires.resize(capacity);
+
+    for (uint32_t i = 0; i < len; i++) {
+      auto &cref = _wires[i];
+      if (!cref) {
+        cref = _pool->acquire(_composer, context);
+        cref->wire->warmup(context);
+        cref->done = false;
+      }
+
+      const SHVar &inputElement = input.payload.seqValue.elements[i];
+
+      // Run within the root flow
+      auto runRes = runSubWire(cref->wire.get(), context, inputElement);
+      if (unlikely(runRes.state == SHRunWireOutputState::Failed)) {
+        // When an error happens during inline execution, propagate the error to the parent wire
+        SHLOG_ERROR("Wire {} failed", wire->name);
+        context->cancelFlow("Wire failed");
+        return cref->wire->previousOutput; // doesn't matter actually
+      } else {
+        // we don't want to propagate a (Return)
+        if (unlikely(runRes.state == SHRunWireOutputState::Stopped)) {
+          context->continueFlow();
+        }
+      }
+
+      // this can be anything really...
+      cloneVar(_outputs[i], cref->wire->previousOutput);
```

Comment:
~In `StepMany`, this line is `_outputs[i] = cref->wire->previousOutput`.
Any reason why `cloneVar()` is necessary here but in `StepMany`?~

nvm: it was the fix in #594